qiskit_machine_learning/neural_networks/circuit_qnn.py

```diff
@@ -375,16 +375,12 @@ def _sample(
     def _probabilities(
         self, input_data: Optional[np.ndarray], weights: Optional[np.ndarray]
     ) -> Union[np.ndarray, SparseArray]:
-
-        if self._quantum_instance is None:
-            raise QiskitMachineLearningError(
-                "Evaluation of probabilities requires a quantum instance!"
-            )
+        self._check_quantum_instance("probabilities")
 
         # evaluate operator
         circuits = []
-        rows = input_data.shape[0]
-        for i in range(rows):
+        num_samples = input_data.shape[0]
+        for i in range(num_samples):
             param_values = {
                 input_param: input_data[i, j] for j, input_param in enumerate(self._input_params)
             }
```

Comment:
Code duplication with lines 352-356.

Reply:
I'd suggest to review implementations of QNNs in general, there are a lot of things to be improved.